featuretools/primitives/base/primitive_base.py

```diff
@@ -45,32 +56,26 @@ def get_filepath(self, filename):
         return os.path.join(config.get("primitive_data_folder"), filename)
 
     def get_args_string(self):
-        # getargspec fails for non user defined inits in python2
-        from sys import version_info
-        if version_info.major < 3:
-            try:
-                parameter_names, _, _, default_values = inspect.getargspec(self.__init__)
-            except TypeError:
-                return ""
-        else:
-            parameter_names, _, _, default_values = inspect.getargspec(self.__init__)
-
-        if default_values is None:
-            default_values = []
-
-        num_kwargs = len(default_values)
-        args = parameter_names[1:-num_kwargs]
-        kwargs = parameter_names[num_kwargs + 1:]
+        if not isinstance(self.__init__, types.MethodType):  # __init__ must be defined
+            return ''
 
-        parameter_format = "{0}={1}"
-        variables = self.__dict__
-        arg_strings = [parameter_format.format(x, variables[x]) for x in args]
+        v2 = version_info.major == 2
+        module = __import__('funcsigs' if v2 else 'inspect')
+        args = module.signature(self.__class__).parameters.values()
 
-        for kwarg, default in zip(kwargs, default_values):
-            val = variables[kwarg]
-            # Only add string where val != default
-            # Handles case where val = np.nan and default = np.nan
-            if val != default and not (np.isnan(val) and np.isnan(default)):
-                arg_strings.append(parameter_format.format(kwarg, val))
+        def valid(arg):
+            error = '"{}" must be attribute of {}'
+            assert hasattr(self, arg.name), error.format(arg.name, self.__class__.__name__)
+            is_positional_or_keyword = arg.kind == arg.POSITIONAL_OR_KEYWORD
```

Comment:
Creating primitives using `make_agg_primitive` and `make_trans_primitive` will return an empty `args_string` because arguments are passed into `__init__` as `**kwargs`.